diff --git c/g_cmds.c w/g_cmds.c
index d2de7f7..581403e 100644
--- c/g_cmds.c
+++ w/g_cmds.c
@@ -724,6 +724,9 @@ void Cmd_Inven_f (edict_t *ent)
 	ent->client->showmod = false;
 	ent->client->showmenu = false;
 	cl->showsquadboard = false; // ADC
+	ent->client->showstatboard = false; // BUZZKILL
+	ent->client->showteamstatboard = false; // BUZZKILL
+	ent->client->showrailboard = false; // BUZZKILL
 
 	if (cl->showinventory)
 	{
@@ -960,6 +963,10 @@ void Cmd_PutAway_f (edict_t *ent)
 	ent->client->showmod = false;
 	ent->client->showmenu = false;
 	ent->client->showsquadboard = false; // ADC
+	ent->client->showstatboard = false; // BUZZKILL
+	ent->client->showteamstatboard = false; // BUZZKILL
+	ent->client->showrailboard = false; // BUZZKILL
+
 	//ent->client->awayframe = level.framenum;
 	gi.WriteByte (svc_layout);
 	gi.WriteString ("");
@@ -2339,6 +2346,23 @@ void ClientCommand(edict_t* ent)
 		return;
 	}
 	// ADC
+	// BUZZKILL
+	if (Q_stricmp(cmd, "playerstats") == 0)
+	{
+		Cmd_Statboard_f(ent);
+		return;
+	}
+	if (Q_stricmp(cmd, "teamstats") == 0)
+	{
+		Cmd_TeamStatboard_f(ent);
+		return;
+	}
+	if (Q_stricmp(cmd, "railstats") == 0)
+	{
+		Cmd_Railboard_f(ent);
+		return;
+	}
+	// BUZZKILL
 	if (Q_stricmp(cmd, "help") == 0)
 	{
 		Cmd_Help_f(ent);
diff --git c/g_combat.c w/g_combat.c
index 477a8f4..39f9bb6 100644
--- c/g_combat.c
+++ w/g_combat.c
@@ -78,6 +78,9 @@ void Killed (edict_t *targ, edict_t *inflictor, edict_t *attacker, int damage, v
 
 	targ->enemy = attacker;
 
+	if (attacker && attacker->client && Q_stricmp(attacker->client->pers.weapon->classname, "weapon_railgun") == 0)
+		stats_add(attacker, STATS_RAIL_KILL, 1); // BUZZKILL - RAIL STATS
+
 #ifdef MONSTERS_OK
 	if ((targ->svflags & SVF_MONSTER) && (targ->deadflag != DEAD_DEAD))
 	{
@@ -399,9 +402,9 @@ qboolean CheckTeamDamage (edict_t *targ, edict_t *attacker)
 	return false;
 }
 
-void T_Damage (edict_t *targ, edict_t *inflictor, edict_t *attacker, vec3_t dir, vec3_t point, vec3_t normal, int damage, int knockback, int dflags, int mod)
+void T_Damage(edict_t* targ, edict_t* inflictor, edict_t* attacker, vec3_t dir, vec3_t point, vec3_t normal, int damage, int knockback, int dflags, int mod)
 {
-	gclient_t	*client;
+	gclient_t* client;
 	int			take;
 	int			vampdrain;
 	int			save;
@@ -411,7 +414,7 @@ void T_Damage (edict_t *targ, edict_t *inflictor, edict_t *attacker, vec3_t dir,
 
 	if (Match_InCountdown())
 		return;
-	
+
 	if (!targ->takedamage)
 		return;
 
@@ -420,12 +423,12 @@ void T_Damage (edict_t *targ, edict_t *inflictor, edict_t *attacker, vec3_t dir,
 	// knockback still occurs
 	if ((targ != attacker) && (deathmatch->value || coop->value))
 	{
-		if (OnSameTeam (targ, attacker))
+		if (OnSameTeam(targ, attacker))
 		{
-			if (	
-					(   ((int)(dmflags->value)) & DF_NO_FRIENDLY_FIRE   )  && 
-					!(dflags & DAMAGE_NO_PROTECTION) &&
-					!(mod & MOD_TELEFRAG)   
+			if (
+				(((int)(dmflags->value)) & DF_NO_FRIENDLY_FIRE) &&
+				!(dflags & DAMAGE_NO_PROTECTION) &&
+				!(mod & MOD_TELEFRAG)
 				)
 			{
 				damage = 0;
@@ -453,7 +456,7 @@ void T_Damage (edict_t *targ, edict_t *inflictor, edict_t *attacker, vec3_t dir,
 
 	VectorNormalize(dir);
 
-// bonus damage for suprising a monster
+	// bonus damage for suprising a monster
 	if (!(dflags & DAMAGE_RADIUS) && (targ->svflags & SVF_MONSTER) && (attacker->client) && (!targ->enemy) && (targ->health > 0))
 		damage *= 2;
 
@@ -463,7 +466,7 @@ void T_Damage (edict_t *targ, edict_t *inflictor, edict_t *attacker, vec3_t dir,
 	// apply Damage rune
 	damage = DamageRuneHook(targ, inflictor, attacker, damage, knockback, dflags);
 
-// figure momentum add
+	// figure momentum add
 	if (!(dflags & DAMAGE_NO_KNOCKBACK))
 	{
 		if ((knockback) && (targ->movetype != MOVETYPE_NONE) && (targ->movetype != MOVETYPE_BOUNCE) && (targ->movetype != MOVETYPE_PUSH) && (targ->movetype != MOVETYPE_STOP))
@@ -476,18 +479,18 @@ void T_Damage (edict_t *targ, edict_t *inflictor, edict_t *attacker, vec3_t dir,
 			else
 				mass = targ->mass;
 
-			if (targ->client  && attacker == targ)
-				VectorScale (dir, 1600.0 * (float)knockback / mass, kvel);	// the rocket jump hack...
+			if (targ->client && attacker == targ)
+				VectorScale(dir, 1600.0 * (float)knockback / mass, kvel);	// the rocket jump hack...
 			else
-				VectorScale (dir, 500.0 * (float)knockback / mass, kvel);
-			
+				VectorScale(dir, 500.0 * (float)knockback / mass, kvel);
+
 #ifdef WEAP_BALANCE_OK	
 			if ((int)ctfflags->value & CTF_WEAP_BALANCE) //a little extra umph to knockback
 			{
-				if (targ->client  && attacker == targ)
-					VectorScale (dir, 1800.0 * (float)knockback / mass, kvel);	// the rocket jump hack...
+				if (targ->client && attacker == targ)
+					VectorScale(dir, 1800.0 * (float)knockback / mass, kvel);	// the rocket jump hack...
 				else
-					VectorScale (dir, 800.0 * (float)knockback / mass, kvel);
+					VectorScale(dir, 800.0 * (float)knockback / mass, kvel);
 			}
 #endif
 			if (targ->groundentity)
@@ -495,7 +498,7 @@ void T_Damage (edict_t *targ, edict_t *inflictor, edict_t *attacker, vec3_t dir,
 				if (kvel[2] < 0)
 					kvel[2] = 0;
 			}
-			VectorAdd (targ->velocity, kvel, targ->velocity);
+			VectorAdd(targ->velocity, kvel, targ->velocity);
 		}
 	}
 
@@ -503,15 +506,15 @@ void T_Damage (edict_t *targ, edict_t *inflictor, edict_t *attacker, vec3_t dir,
 	save = 0;
 
 	// check for godmode
-	if ( (targ->flags & FL_GODMODE) && !(dflags & DAMAGE_NO_PROTECTION) )
+	if ((targ->flags & FL_GODMODE) && !(dflags & DAMAGE_NO_PROTECTION))
 	{
 		take = 0;
 		save = damage;
-		SpawnDamage (te_sparks, point, normal, save);
+		SpawnDamage(te_sparks, point, normal, save);
 	}
 
 	// check for invincibility
-	if ((client && client->invincible_framenum > level.framenum ) && !(dflags & DAMAGE_NO_PROTECTION))
+	if ((client && client->invincible_framenum > level.framenum) && !(dflags & DAMAGE_NO_PROTECTION))
 	{
 		if (targ->pain_debounce_time < level.time)
 		{
@@ -522,30 +525,45 @@ void T_Damage (edict_t *targ, edict_t *inflictor, edict_t *attacker, vec3_t dir,
 		save = damage;
 	}
 
-	psave = CheckPowerArmor (targ, point, normal, take, dflags);
+	psave = CheckPowerArmor(targ, point, normal, take, dflags);
 	take -= psave;
 
 	// apply Resist rune
 	take = ResistRuneHook(targ, inflictor, attacker, take, knockback, dflags);
 
 	// armor protect from teammates...only protect normal armor
-	if (((int) ctfflags->value & CTF_TEAM_ARMOR_PROTECT) &&
-	    (OnSameTeam (targ, attacker)) &&
-	    (targ != attacker))
-	  asave = 0;
+	if (((int)ctfflags->value & CTF_TEAM_ARMOR_PROTECT) &&
+		(OnSameTeam(targ, attacker)) &&
+		(targ != attacker))
+		asave = 0;
 	else
 	{
-	  asave = CheckArmor (targ, point, normal, take, te_sparks, dflags);
-	  take -= asave;
+		asave = CheckArmor(targ, point, normal, take, te_sparks, dflags);
+		take -= asave;
 	}
 
 	//treat cheat/powerup savings the same as armor
 	asave += save;
 
 	// team damage avoidance
-	if (!(dflags & DAMAGE_NO_PROTECTION) && CheckTeamDamage (targ, attacker))
+	if (!(dflags & DAMAGE_NO_PROTECTION) && CheckTeamDamage(targ, attacker))
 		return;
 
+	// BUZZKILL - ADVANCED ANALYTICS - START
+	if (attacker && attacker->client)
+	{
+		if (Q_stricmp(attacker->client->pers.weapon->classname, "weapon_railgun") == 0)
+		{
+			stats_add(attacker, STATS_RAIL_HIT, 1);
+		}
+		/*if (targ && targ->client && damage && mod != MOD_TELEFRAG)
+		{
+			stats_add(attacker, STATS_DAMAGE_GIVEN, damage);
+			stats_add(targ, STATS_DAMAGE_REC, damage);
+		}*/
+	}
+	// BUZZKILL - ADVANCED ANALYTICS - END
+
 // do the damage
 	if (take)
 	{
@@ -554,7 +572,6 @@ void T_Damage (edict_t *targ, edict_t *inflictor, edict_t *attacker, vec3_t dir,
 		else
 			SpawnDamage (te_sparks, point, normal, take);
 
-
 		targ->health = targ->health - take;
 
 		if (attacker && attacker->client && attacker->client->rune)   //added by Vampire - if the attacker is this client
@@ -593,6 +610,7 @@ void T_Damage (edict_t *targ, edict_t *inflictor, edict_t *attacker, vec3_t dir,
 		{
 			if (attacker && attacker->client)
 				attacker->client->hit_carrier_time = level.time;
+				gi.sound(attacker, CHAN_ITEM, gi.soundindex("fc-hit.wav"), 1, ATTN_NORM, 0);
 		}
 		// END LM_JORM
 			
diff --git c/g_ctffunc.c w/g_ctffunc.c
index ac8afd1..e912f5b 100644
--- c/g_ctffunc.c
+++ w/g_ctffunc.c
@@ -5,6 +5,7 @@
 #include "p_stats.h"
 #include "gslog.h"
 #include "bat.h"
+#include <time.h>
 
 edict_t * ctf_findplayer(edict_t * ent_after, edict_t * ignore, int teamnum_wanted)
 {
diff --git c/g_ctffunc.h w/g_ctffunc.h
index 47f2eef..f14e705 100644
--- c/g_ctffunc.h
+++ w/g_ctffunc.h
@@ -1,7 +1,7 @@
 //#define CTF_TEAM_LIMIT			3 //you can do loops less than this
 //#define CTF_TEAM_BLUE			2
 //#define CTF_TEAM_RED			1
-//#define CTF_TEAM_UNDEFINED		0
+//#define CTF_TEAM_UNDEFINED	0
 //#define CTF_TEAM_OBSERVER		-1
 
 //-----making teams for observers-----
@@ -115,7 +115,5 @@ void ctf_ChangeMap(char *mapname, qboolean startmatch);
 
 //stuff currently in other files but needed 
 int ClientShowID(edict_t *ent, char * buf);
-
 void CTFSquadboardMessage (edict_t *ent, edict_t *killer); // ADC
-
 void ctf_PassItem(edict_t* startent, edict_t* tossent);
diff --git c/g_local.h w/g_local.h
index 9cc0471..4f42970 100644
--- c/g_local.h
+++ w/g_local.h
@@ -14,6 +14,8 @@ _CrtMemState startup1;	// memory diagnostics
 #define OutputDebugString	//not doing Windows
 #endif
 
+#include <mysqlx/xapi.h>
+
 //#define ZBOT
 
 //#define OLDOBSERVERCODE
@@ -658,6 +660,9 @@ void Cmd_WeapNext_f(edict_t *ent);
 void PlayTeamSound(edict_t *ent, char *sound); // LM_JORM
 void PlayVoiceSound(edict_t *ent, char *sound); // LM_Surt
 void Cmd_Squadboard_f (edict_t *ent); // ADC
+void Cmd_Statboard_f(edict_t* ent); // BUZZKILL
+void Cmd_TeamStatboard_f(edict_t* ent); // BUZZKILL
+void Cmd_Railboard_f(edict_t* ent); // BUZZKILL
 
 //
 // g_items.c
@@ -889,6 +894,9 @@ void G_CheckChaseStats (edict_t *ent);
 void ValidateSelectedItem (edict_t *ent);
 void DeathmatchScoreboardMessage (edict_t *client, edict_t *killer);
 void SquadboardMessage (edict_t *client, edict_t *killer); // ADC
+void StatboardMessage(edict_t* client, edict_t* killer); // BUZZKILL
+void TeamStatboardMessage(edict_t* client, edict_t* killer); // BUZZKILL
+void RailboardMessage(edict_t* client, edict_t* killer); // BUZZKILL
 
 //
 // g_pweapon.c
@@ -1028,6 +1036,38 @@ typedef struct
 	//qboolean						toss_state; // BUZZKILL - should tossed thing track nearest player?
 } client_ctf_t;
 
+typedef struct playerstats_s
+{
+	int administrator;
+
+	unsigned int frags;
+	unsigned int fragged;
+
+	unsigned long shots;
+	unsigned long shots_hit;
+
+	unsigned int num_sprees;
+	int max_streak;
+	int suicides;
+
+	//ctf
+	int flag_pickups;	//number of times player grabs the flag
+	int flag_captures;	//number of times player caps the flag
+	int flag_returns;	//number of times player returns his own flag
+	int flag_kills;		//number of times player kills the flag carrier
+	int offense_kills;	//number of times player kills a defender
+	int defense_kills;	//number of times player defends his base by killing a player
+	int assists;		//number of times player gains an assist (flag_kill or flag_return just before a flag_cap)
+	//end ctf
+
+	int playingtime;			//Playing time today (in seconds)
+	int total_playtime;			//Total playing time in minutes
+
+	char member_since[30];
+	char last_played[30];
+	char player_name[24];
+
+}playerstats_t;
 
 // this structure is cleared on each PutClientInServer(),
 // except for 'client->pers'
@@ -1046,7 +1086,10 @@ struct gclient_s
 	qboolean	showinventory;		// set layout stat
 	qboolean	showhelp;
 	qboolean	showhelpicon;
-	qboolean		showsquadboard;	// ADC
+	qboolean	showsquadboard;	// ADC
+	qboolean	showstatboard; //BUZZKILL
+	qboolean	showteamstatboard; //BUZZKILL
+	qboolean	showrailboard; //BUZZKILL
 
 	int			ammo_index;
 
@@ -1326,6 +1369,10 @@ struct edict_s
 	float           droptime;
 	int             entprops; //flags to tag entities with, for use with flags, which have no client
 	// END CTF CODE
+
+	// BUZZKILL -- SQL (TEMP)
+	playerstats_t ctfstats;
 };
 
+
 #endif
diff --git c/g_menu.c w/g_menu.c
index 42d973f..6675d41 100644
--- c/g_menu.c
+++ w/g_menu.c
@@ -247,6 +247,9 @@ void Ctf_Menu (edict_t *ent)
 	cl->showmod = false;
 	cl->showctfhud = false;
 	cl->showsquadboard = false; // ADC
+	cl->showstatboard = false; // BUZZKILL
+	cl->showteamstatboard = false; // BUZZKILL
+	cl->showrailboard = false; // BUZZKILL
 
 	if (cl->showmenu)
 	{
@@ -567,6 +570,9 @@ void Toggle_Radio_Menu (edict_t *ent)
 	cl->showmod = false;
 	cl->showctfhud = false;
 	cl->showsquadboard = false; // ADC
+	cl->showstatboard = false; // BUZZKILL
+	cl->showteamstatboard = false; // BUZZKILL
+	cl->showrailboard = false; // BUZZKILL
 
 	if (cl->showmenu)
 	{
@@ -680,6 +686,9 @@ void Ref_Main_Load (edict_t *ent)
 	cl->showmod = false;
 	cl->showctfhud = false;
 	cl->showsquadboard = false; // ADC
+	cl->showstatboard = false; // BUZZKILL
+	cl->showteamstatboard = false; // BUZZKILL
+	cl->showrailboard = false; // BUZZKILL
 
 	if (cl->showmenu)
 	{
diff --git c/g_runes.c w/g_runes.c
index 68074f3..b62e98e 100644
--- c/g_runes.c
+++ w/g_runes.c
@@ -444,7 +444,8 @@ void SP_damage_rune(edict_t* self)
 qboolean Pickup_Rune(edict_t* ent, edict_t* other)
 {
 	char* runeName = NULL; // BUZZKILL - IMPROVED ANALYTICS
-	int		runeID = 0; // BUZZKILL - IMPROVED ANALYTICS // problematic if this remains at 0
+	int		runeID = 0; // BUZZKILL - IMPROVED ANALYTICS
+	int		runeHas = 0;
 
 	if (!other->client->rune) // don't already have rune
 	{
@@ -466,24 +467,29 @@ qboolean Pickup_Rune(edict_t* ent, edict_t* other)
 		case RUNE_DAMAGE:
 			runeName = "rune_damage";
 			runeID = STATS_RUNE_STRENGTH;
+			runeHas = STATS_HAS_ST;
 			break;
 		case RUNE_HASTE:
 			runeName = "rune_haste";
 			runeID = STATS_RUNE_HASTE;
+			runeHas = STATS_HAS_HA;
 			break;
 		case RUNE_REGEN:
 			runeName = "rune_regen";
 			runeID = STATS_RUNE_REGEN;
+			runeHas = STATS_HAS_RG;
 			break;
 		case RUNE_RESIST:
 			runeName = "rune_resist";
 			runeID = STATS_RUNE_RESIST;
+			runeHas = STATS_HAS_RS;
 			break;
 		default:
 			gi.dprintf("Bad rune selected.\n");
 			break;
 		}
 		stats_add(other, runeID, 1);
+		stats_set(other, runeHas, 1);
 		sl_LogPickup(&gi,
 			"PlayerEvent",
 			other->client->pers.netname,
@@ -581,6 +587,7 @@ void homing_think(edict_t* ent)
 	}
 	ent->touch = Touch_Item;
 	ent->nextthink = level.time + FRAMETIME;
+	
 }
 // BUZZKILL - TOSS THING - END
 
@@ -590,6 +597,7 @@ void Drop_Rune_Think(edict_t* ent)
 	ent->owner = NULL;
 	ent->think = Rune_Think;
 	ent->nextthink = level.time + FRAMETIME;
+
 }
 
 extern void drop_temp_touch(edict_t* ent, edict_t* other, cplane_t* plane, csurface_t* surf);
@@ -604,6 +612,11 @@ void Drop_Rune(edict_t* ent, gitem_t* item)
 
 	ent->client->pers.inventory[ITEM_INDEX(item)]--;
 
+	stats_set(ent, STATS_HAS_ST, 0);
+	stats_set(ent, STATS_HAS_RS, 0);
+	stats_set(ent, STATS_HAS_HA, 0);
+	stats_set(ent, STATS_HAS_RG, 0);
+
 	dropped = ent->client->rune;
 	if (!dropped)
 		return;
@@ -649,6 +662,11 @@ void Toss_Rune(edict_t* ent, gitem_t* item)
 
 	ent->client->pers.inventory[ITEM_INDEX(item)]--;
 
+	stats_set(ent, STATS_HAS_ST, 0);
+	stats_set(ent, STATS_HAS_RS, 0);
+	stats_set(ent, STATS_HAS_HA, 0);
+	stats_set(ent, STATS_HAS_RG, 0);
+
 	dropped = ent->client->rune;
 	if (!dropped)
 		return;
diff --git c/g_weapon.c w/g_weapon.c
index 0e95d2a..c0ce7a8 100644
--- c/g_weapon.c
+++ w/g_weapon.c
@@ -772,6 +772,7 @@ void fire_rail (edict_t *self, vec3_t start, vec3_t aimdir, int damage, int kick
 
 	if (self->client)
 		PlayerNoise(self, tr.endpos, PNOISE_IMPACT);
+		stats_add(self, STATS_RAIL_SHOT, 1); // BUZZKILL - RAIL STATS
 }
 
 
diff --git c/p_hud.c w/p_hud.c
index fdc1981..24e91b0 100644
--- c/p_hud.c
+++ w/p_hud.c
@@ -182,8 +182,8 @@ void DeathmatchScoreboardMessage (edict_t *ent, edict_t *killer)
 	int     bluescore, redscore;  // TEAM PLAY -- LM_JORM
     int     bluecaps, redcaps;  // TEAM PLAY -- LM_JORM
     int     blue, red;  // TEAM PLAY -- LM_JORM
-    int     bluepings, redpings;  //bat
-    int     Avg_Bluepings, Avg_Redpings;  //bat
+    blue = 0;
+    red = 0;
 
     // BUZZKILL - ADVANCED ANALYTICS SCOREBOARD - START
     int     blue_rune_strength = 0;
@@ -206,6 +206,10 @@ void DeathmatchScoreboardMessage (edict_t *ent, edict_t *killer)
     int     rfctest = 0;
     char*   redfc = NULL;
     char*   bluefc = NULL;
+    char*   red_runes = NULL;
+    char*   blue_runes = NULL;
+    int     red_rune_acc = 0;
+    int     blue_rune_acc = 0;
     // BUZZKILL - ADVANCED ANALYTICS SCOREBOARD - END
 
     size_t  stringlength;
@@ -213,7 +217,7 @@ void DeathmatchScoreboardMessage (edict_t *ent, edict_t *killer)
     size_t  j;
     int     k;
     int     l;
-    //int       sortedscores[MAX_CLIENTS];
+    
     int     redsorted[MAX_CLIENTS];
     int     redsortedscores[MAX_CLIENTS];
     int     bluesorted[MAX_CLIENTS];
@@ -229,9 +233,8 @@ void DeathmatchScoreboardMessage (edict_t *ent, edict_t *killer)
     int     reg_observers = 0;
     int     red_observers = 0;
     int     blue_observers = 0;
+    int     total_observers = 0;
 
-    
-	//int     score, total;
     int     score;
 
     // BUZZKILL - ADVANCED ANALYTICS SCOREBOARD - START
@@ -243,13 +246,15 @@ void DeathmatchScoreboardMessage (edict_t *ent, edict_t *killer)
     int     item_armor;
     int     item_shield;
     int     item_quad;
+    char*   player_rune;
+
+    player_rune = NULL;
     // BUZZKILL - ADVANCED ANALYTICS SCOREBOARD - END
 
- //   int     picnum;
     int     x, y;
     gclient_t   *cl;
     edict_t     *cl_ent;
-    char    *tag;
+
     qboolean    showsmall;
     qboolean    is_red_fc;
     qboolean    is_blue_fc;
@@ -257,7 +262,6 @@ void DeathmatchScoreboardMessage (edict_t *ent, edict_t *killer)
     is_red_fc = false;
     is_blue_fc = false;
     showsmall = false;
-//    showsmall = true;
 
     bluescore = bluecaps = blue = 0; // TEAM PLAY -- LM_JORM
     redscore = redcaps = red = 0;  // TEAM PLAY -- LM_JORM
@@ -265,20 +269,7 @@ void DeathmatchScoreboardMessage (edict_t *ent, edict_t *killer)
     observers = 0;  // TEAM PLAY -- LM_JORM
 #endif
 
-	//bat
-	redpings = 0;
-	bluepings = 0;
-
-
-
-	//
-	// This function really should be rewritten, but I probably 
-	// won't get around to it.
-	// -bat
-	//
-
 	// sort the clients by score
-    //total = 0;
     for (i=0; i<game.maxclients; i++)
     {
         cl_ent = g_edicts + 1 + i;
@@ -308,13 +299,13 @@ void DeathmatchScoreboardMessage (edict_t *ent, edict_t *killer)
         item_shield = stats_get(cl_ent, STATS_ITEM_SHIELD);
         item_armor = stats_get(cl_ent, STATS_ITEM_ARMOR);
         item_mega = stats_get(cl_ent, STATS_ITEM_MEGA);
+        
         // BUZZKILL - ADVANCED ANALYTICS SCOREBOARD - END
 
         if (cl_ent->client->ctf.teamnum == CTF_TEAM_RED) // RED TEAM
         {
             redscore += score;
             redcaps += stats_get(cl_ent, STATS_CAPTURES); 
-			redpings += cl->ping;
 
             // BUZZKILL - ADVANCED ANALYTICS SCOREBOARD - START
             red_rune_strength += rune_strength;
@@ -326,10 +317,39 @@ void DeathmatchScoreboardMessage (edict_t *ent, edict_t *killer)
             red_item_armor += item_armor;
             red_item_mega += item_mega;
 
+            if (cl->rune)
+            {
+                red_rune_acc += cl->rune->runetype;
+            }
+            else
+            {
+                red_rune_acc += 0;
+            }
+
             is_red_fc = stats_get(cl_ent, STATS_IS_FC);
 
             if (is_red_fc)
                 redfc = cl_ent->client->pers.netname;
+
+            switch (red_rune_acc)
+            {
+                case 1:	 red_runes = "ST";    break;
+                case 2:	 red_runes = "RS";    break;
+                case 3:	 red_runes = "ST RS";    break;
+                case 4:	 red_runes = "HA";    break;
+                case 5:	 red_runes = "ST HA";    break;
+                case 6:	 red_runes = "RS HA";    break;
+                case 7:	 red_runes = "ST RS HA";    break;
+                case 8:	 red_runes = "RG";    break;
+                case 9:	 red_runes = "ST RG";    break;
+                case 10: red_runes = "RS RG";    break;
+                case 11: red_runes = "ST RS RG";    break;
+                case 12: red_runes = "HA RG";    break;
+                case 13: red_runes = "ST HA RG";    break;
+                case 14: red_runes = "RS HA RG";    break;
+                case 15: red_runes = "ST RS HA RG";    break;
+                default: red_runes = " ";    break;
+            }
             // BUZZKILL - ADVANCED ANALYTICS SCOREBOARD - END
 
             for (j=0 ; j<red ; j++)
@@ -356,7 +376,6 @@ void DeathmatchScoreboardMessage (edict_t *ent, edict_t *killer)
         {
             bluescore += score;
             bluecaps  += stats_get(cl_ent, STATS_CAPTURES);
-			bluepings += cl->ping;
 
             // BUZZKILL - ADVANCED ANALYTICS SCOREBOARD - START
             blue_rune_strength += rune_strength;
@@ -367,11 +386,40 @@ void DeathmatchScoreboardMessage (edict_t *ent, edict_t *killer)
             blue_item_shield += item_shield;
             blue_item_armor += item_armor;
             blue_item_mega += item_mega;
-            
+
+            if (cl->rune)
+            {
+                blue_rune_acc += cl->rune->runetype;
+            }
+            else
+            {
+                blue_rune_acc += 0;
+            }
+
             is_blue_fc = stats_get(cl_ent, STATS_IS_FC);
 
             if (is_blue_fc)
                 bluefc = cl_ent->client->pers.netname;
+
+            switch(blue_rune_acc)
+            {
+                case 1:	 blue_runes = "ST";    break;
+                case 2:	 blue_runes = "RS";    break;
+                case 3:	 blue_runes = "ST RS";    break;
+                case 4:	 blue_runes = "HA";    break;
+                case 5:	 blue_runes = "ST HA";    break;
+                case 6:	 blue_runes = "RS HA";    break;
+                case 7:	 blue_runes = "ST RS HA";    break;
+                case 8:	 blue_runes = "RG";    break;
+                case 9:	 blue_runes = "ST RG";    break;
+                case 10: blue_runes = "RS RG";    break;
+                case 11: blue_runes = "ST RS RG";    break;
+                case 12: blue_runes = "HA RG";    break;
+                case 13: blue_runes = "ST HA RG";    break;
+                case 14: blue_runes = "RS HA RG";    break;
+                case 15: blue_runes = "ST RS HA RG";    break;
+                default: blue_runes = " ";    break;
+            }
             // BUZZKILL - ADVANCED ANALYTICS SCOREBOARD - END
 
             for (j=0 ; j<blue ; j++)
@@ -398,11 +446,13 @@ void DeathmatchScoreboardMessage (edict_t *ent, edict_t *killer)
 		{
 			sorted_blue_observers[blue_observers] = i;
 			blue_observers++;
+            total_observers++;
 		}
         else if (cl_ent->client->ctf.teamnum == CTF_TEAM_OBSERVER_RED)
 		{
 			sorted_red_observers[red_observers] = i;
 			red_observers++;
+            total_observers++;
 		}
 //bat - put this back in
 //#ifdef OLDOBSERVERCODE
@@ -410,6 +460,7 @@ void DeathmatchScoreboardMessage (edict_t *ent, edict_t *killer)
 		{
 			sorted_reg_observers[reg_observers] = i;
 			reg_observers++;
+            total_observers++;
 		}
 //#endif
         //total++;
@@ -439,31 +490,20 @@ void DeathmatchScoreboardMessage (edict_t *ent, edict_t *killer)
 
 
     // add the clients in sorted order
-    if (red > 6)
+    if (red > 6 || red + blue + total_observers > 16)
     {
         showsmall = true;
         if (red > 21)
             red = 21;
     }
     
-	if (blue > 6)
+	if (blue > 6 || red + blue + total_observers > 16)
     {
         showsmall = true;
         if (blue > 21)
             blue = 21;
     }
-
    
-	if(red == 0)
-		Avg_Redpings = 0;
-	else
-		Avg_Redpings = redpings/red;
-	
-	if(blue == 0)
-		Avg_Bluepings =	0;
-	else
-		Avg_Bluepings = bluepings/blue;
-	
     if (showsmall)
     {
         x = 0;
@@ -495,6 +535,26 @@ void DeathmatchScoreboardMessage (edict_t *ent, edict_t *killer)
             x = 0;
             y = 48 + 8 * i;
 
+            if (cl->rune)
+            {
+                switch (cl->rune->runetype)
+                {
+                case 1:	 player_rune = "ST";    break;
+                case 2:	 player_rune = "RS";    break;
+                case 4:	 player_rune = "HA";    break;
+                case 8:	 player_rune = "RG";    break;
+                }
+                Com_sprintf(string2, sizeof(string2),
+                    "xv %i yv %i string2 \"%s\" ",
+                    x + 32 - 136 + 80, y, player_rune);
+
+                j = strlen(string2);
+                if (stringlength + j > 1024)
+                    break;
+                strcpy(string + stringlength, string2);
+                stringlength += j;
+            }
+
             if (cl_ent == Query_DMVP())
 			{
 				sprintf(mvpstring, "D%3d %3d %s", cl->resp.score, cl->ping, cl->pers.netname);
@@ -521,10 +581,11 @@ void DeathmatchScoreboardMessage (edict_t *ent, edict_t *killer)
         }
         else
         {
- //           picnum = gi.imageindex ("i_fixme");
+            //picnum = gi.imageindex ("i_fixme");
             x = 0;
             y = 32 + 32 * (i%6);
-
+            //tag = NULL;
+        /*
             // add a dogtag
             if (cl_ent == ent)
                 tag = "tag1";
@@ -542,13 +603,14 @@ void DeathmatchScoreboardMessage (edict_t *ent, edict_t *killer)
                 strcpy (string + stringlength, entry);
                 stringlength += j;
             }
+        */
 
         // send the layout
-            Com_sprintf (entry, sizeof(entry),
-                "client %i %i %i %i %i %i ",
-                x, y, redsorted[i], stats_get(cl_ent, STATS_SCORE), 
-                cl->ping,
-                (level.framenum - cl->resp.enterframe)/600);    
+
+        Com_sprintf(entry, sizeof(entry),
+            "client %i %i %i %i %i %i ",
+            x, y, redsorted[i], stats_get(cl_ent, STATS_SCORE),
+            cl->ping, (level.framenum - cl->resp.enterframe) / 600);
 
             j = strlen(entry);
             if (stringlength + j > 1024)
@@ -570,6 +632,26 @@ void DeathmatchScoreboardMessage (edict_t *ent, edict_t *killer)
                 stringlength += j;
             }
 
+            if (cl->rune)
+            {
+                switch (cl->rune->runetype)
+                {
+                    case 1:	 player_rune = "ST";    break;
+                    case 2:	 player_rune = "RS";    break;
+                    case 4:	 player_rune = "HA";    break;
+                    case 8:	 player_rune = "RG";    break;
+                }
+                Com_sprintf(string2, sizeof(string2),
+                    "xv %i yv %i string2 \"R:%s\" ",
+                    x + 32 + 80, y + 16, player_rune);
+
+                j = strlen(string2);
+                if (stringlength + j > 1024)
+                    break;
+                strcpy(string + stringlength, string2);
+                stringlength += j;
+            }
+
             if (cl_ent == Query_DMVP())
             {
                 Com_sprintf (string2, sizeof(string2),
@@ -608,6 +690,26 @@ void DeathmatchScoreboardMessage (edict_t *ent, edict_t *killer)
             x = 160;
             y = 48 + 8 * i;
 
+            if (cl->rune)
+            {
+                switch (cl->rune->runetype)
+                {
+                case 1:	 player_rune = "ST";    break;
+                case 2:	 player_rune = "RS";    break;
+                case 4:	 player_rune = "HA";    break;
+                case 8:	 player_rune = "RG";    break;
+                }
+                Com_sprintf(string2, sizeof(string2),
+                    "xv %i yv %i string2 \"%s\" ",
+                    x + 32 + 56 + 80, y, player_rune);
+
+                j = strlen(string2);
+                if (stringlength + j > 1024)
+                    break;
+                strcpy(string + stringlength, string2);
+                stringlength += j;
+            }
+
             if (cl_ent == Query_DMVP())
 			{
 				sprintf(mvpstring, "D%3d %3d %s", cl->resp.score, cl->ping, cl->pers.netname);
@@ -639,11 +741,12 @@ void DeathmatchScoreboardMessage (edict_t *ent, edict_t *killer)
         }
         else
         {
- //           picnum = gi.imageindex ("i_fixme");
+            //picnum = gi.imageindex ("i_fixme");
             //x = (i>=6) ? 160 : 0;
             x = 160;
             y = 32 + 32 * (i%6);
-
+            //tag = NULL;
+        /*
             // add a dogtag
             if (cl_ent == ent)
                 tag = "tag1";
@@ -661,12 +764,12 @@ void DeathmatchScoreboardMessage (edict_t *ent, edict_t *killer)
                 strcpy (string + stringlength, entry);
                 stringlength += j;
             }
-
+        */
         // send the layout
-            Com_sprintf (entry, sizeof(entry),
+            Com_sprintf(entry, sizeof(entry),
                 "client %i %i %i %i %i %i ",
                 x, y, bluesorted[i], stats_get(cl_ent, STATS_SCORE),
-                cl->ping, (level.framenum - cl->resp.enterframe)/600);  
+                cl->ping, (level.framenum - cl->resp.enterframe) / 600);
 
             j = strlen(entry);
             if (stringlength + j > 1024)
@@ -687,6 +790,28 @@ void DeathmatchScoreboardMessage (edict_t *ent, edict_t *killer)
                 strcpy (string + stringlength, string2);
                 stringlength += j;
             }
+
+            if (cl->rune)
+            {
+                switch (cl->rune->runetype)
+                {
+                    case 1:	 player_rune = "ST";    break;
+                    case 2:	 player_rune = "RS";    break;
+                    case 4:	 player_rune = "HA";    break;
+                    case 8:	 player_rune = "RG";    break;
+                }
+
+                Com_sprintf(string2, sizeof(string2),
+                    "xv %i yv %i string2 \"R:%s\" ",
+                    x + 32 + 80, y + 16, player_rune);
+
+                j = strlen(string2);
+                if (stringlength + j > 1024)
+                    break;
+                strcpy(string + stringlength, string2);
+                stringlength += j;
+            }
+
             if (cl_ent == Query_DMVP())
             {
                 Com_sprintf (string2, sizeof(string2),
@@ -769,21 +894,23 @@ void DeathmatchScoreboardMessage (edict_t *ent, edict_t *killer)
 			x = 0;
 			Show_String(x, y, string2, "Red Observers:");
 			y += 8;
+            if (red + blue + total_observers <= 16)
+            {
+                for (i = 0; i < red_observers; i++)
+                {
+                    cl = &game.clients[sorted_red_observers[i]];
+                    cl_ent = g_edicts + 1 + sorted_red_observers[i];
+                    Show_String(x, y, string2, cl->pers.netname);
+                    y += 8;
+                }
 
-			for (i = 0; i < red_observers ; i++)
-			{
-				cl = &game.clients[sorted_red_observers[i]];
-				cl_ent = g_edicts + 1 + sorted_red_observers[i];  
-				Show_String(x, y, string2, cl->pers.netname);
-				y += 8;
-			}
-
-			j = strlen(string2);
-			if (stringlength + j <= 1024)
-			{
-				strcpy (string + stringlength, string2);
-				stringlength += j;
-			}
+                j = strlen(string2);
+                if (stringlength + j <= 1024)
+                {
+                    strcpy(string + stringlength, string2);
+                    stringlength += j;
+                }
+            }
 		}
 
 		if(blue_observers)
@@ -791,80 +918,85 @@ void DeathmatchScoreboardMessage (edict_t *ent, edict_t *killer)
 			x = 160;
 			Show_String(x, y, string2, "Blue Observers:");
 			y += 8;
+            if (red + blue + total_observers <= 16)
+            {
+                for (i = 0; i < blue_observers; i++)
+                {
+                    cl = &game.clients[sorted_blue_observers[i]];
+                    cl_ent = g_edicts + 1 + sorted_blue_observers[i];
+                    //strcat(entry, cl->pers.netname);
+                    Show_String(x, y, string2, cl->pers.netname);
+                    y += 8;
+                }
 
-			for (i = 0; i < blue_observers ; i++)
-			{
-				cl = &game.clients[sorted_blue_observers[i]];
-				cl_ent = g_edicts + 1 + sorted_blue_observers[i];  
-				//strcat(entry, cl->pers.netname);
-				Show_String(x, y, string2, cl->pers.netname);
-				y += 8;
-			}
-
-			j = strlen(string2);
-			if (stringlength + j <= 1024)
-			{
-				strcpy (string + stringlength, string2);
-				stringlength += j;
-			}
+                j = strlen(string2);
+                if (stringlength + j <= 1024)
+                {
+                    strcpy(string + stringlength, string2);
+                    stringlength += j;
+                }
+            }
 		}
 
 
 
 		if(reg_observers)
 		{
-			//give more space for the reg observers
-			if(red_observers == 0 && blue_observers == 0)
-			{
-				x = 80;
-				Show_String(x, y, string2, "Observers:");
-				y += 8;
+            if (red + blue + total_observers <= 16)
+            {
+                //give more space for the reg observers
+                if (red_observers == 0 && blue_observers == 0)
+                {
+                    x = 80;
+                    Show_String(x, y, string2, "Observers:");
+                    y += 8;
 
-				//Do 2 obs per line
+                    //Do 2 obs per line
 
-				for(i = 0; i < reg_observers ; i++)
-				{
-					//x = (i & 0x01) * 160;
-					x = (i % 3) * 130;
-				
-					cl = &game.clients[sorted_reg_observers[i]];
-					cl_ent = g_edicts + 1 + sorted_reg_observers[i];  
-					Show_String(x, y, string2, cl->pers.netname);
-				
-					if((i % 3) == 2)
-						y += 8;
-				}
+                    for (i = 0; i < reg_observers; i++)
+                    {
+                        //x = (i & 0x01) * 160;
+                        x = (i % 3) * 130;
 
-				j = strlen(string2);
-				if (stringlength + j <= 1024)
-				{
-					strcpy (string + stringlength, string2);
-					stringlength += j;
-				}
-		
-			}
-			else
-			{
-				x = 80;
-				Show_String(x, y, string2, "Observers:");
-				y += 8;
+                        cl = &game.clients[sorted_reg_observers[i]];
+                        cl_ent = g_edicts + 1 + sorted_reg_observers[i];
+                        Show_String(x, y, string2, cl->pers.netname);
 
-				for (i = 0; i < reg_observers ; i++)
-				{
-					cl = &game.clients[sorted_reg_observers[i]];
-					cl_ent = g_edicts + 1 + sorted_reg_observers[i];  
-					//strcat(entry, cl->pers.netname);
-					Show_String(x, y, string2, cl->pers.netname);
-					y += 8;
-				}
+                        if ((i % 3) == 2)
+                            y += 8;
+                    }
 
-				j = strlen(string2);
-				if (stringlength + j <= 1024)
-				{
-					strcpy (string + stringlength, string2);
-					stringlength += j;
-				}
-			}
+                    j = strlen(string2);
+                    if (stringlength + j <= 1024)
+                    {
+                        strcpy(string + stringlength, string2);
+                        stringlength += j;
+                    }
+
+                }
+                else
+                {
+                    x = 80;
+                    Show_String(x, y, string2, "Observers:");
+                    y += 8;
+
+                    for (i = 0; i < reg_observers; i++)
+                    {
+                        cl = &game.clients[sorted_reg_observers[i]];
+                        cl_ent = g_edicts + 1 + sorted_reg_observers[i];
+                        //strcat(entry, cl->pers.netname);
+                        Show_String(x, y, string2, cl->pers.netname);
+                        y += 8;
+                    }
+
+                    j = strlen(string2);
+                    if (stringlength + j <= 1024)
+                    {
+                        strcpy(string + stringlength, string2);
+                        stringlength += j;
+                    }
+                }
+            }
 		}
 	}
 
@@ -915,75 +1047,43 @@ void DeathmatchScoreboardMessage (edict_t *ent, edict_t *killer)
         "xv %i yv %i picn %s "
         "xv %i yv %i picn %s "
 
-        "xv %i yv %i picn %s "
-        "xv %i yv %i picn %s "
-        "xv %i yv %i picn %s "
-        "xv %i yv %i picn %s "
+        //"xv %i yv %i string2 \"CAPS:%i\" "    // Captures // NOT NEEDED - IN v6.1 HUD
+        "xv %i yv %i string2 \"FC:\" "// BUZZKILL - FLAG CARRIER LABEL
+        "xv %i yv %i string2 \"%s\" "           // BUZZKILL - FLAG CARRIER NAME
+        "xv %i yv %i string2 \"Runes\" "       // BUZZKILL - RUNE LISTING
+        "xv %i yv %i string2 \"%s\" "           // BUZZKILL - RUNE LISTING
 
-        "xv %i yv %i string2 \"CAPS:%i\" "      // Captures
-        "xv %i yv %i string2 \"Q:%i\" "         // BUZZKILL - QUAD GRABS BY TEAM
-        "xv %i yv %i string2 \"   S:%i\" "      // BUZZKILL - SHIELD GRABS BY TEAM
-        "xv %i yv %i string2 \"A:%i\" "         // BUZZKILL - RED ARMOR GRABS BY TEAM
-        "xv %i yv %i string2 \"   M:%i\" "      // BUZZKILL - MEGA HEALTH GRABS BY TEAM
-        //"xv %i yv %i string2 \"AP:%3i\" "     // BUZZKILL - does anybody actually care about average ping anymore?
-        
-        "xv %i yv %i string2 \"CAPS:%i\" "
-        "xv %i yv %i string2 \"Q:%i\" "
-        "xv %i yv %i string2 \"   S:%i\" "
-        "xv %i yv %i string2 \"A:%i\" "
-        "xv %i yv %i string2 \"   M:%i\" "
+        "xv %i yv %i string2 \"FC:\" "// BUZZKILL - FLAG CARRIER LABEL
+        "xv %i yv %i string2 \"%s\" "           // BUZZKILL - FLAG CARRIER NAME
+        "xv %i yv %i string2 \"Runes\" "           // BUZZKILL - RUNE LISTING
+        "xv %i yv %i string2 \"%s\" ",    // BUZZKILL - RUNE LISTING
 
-        "xv %i yv %i string2 \"FC:%s\" "        // BUZZKILL - FLAG CARRIER
-        "xv %i yv %i string2 \"ST:%i\" "        // BUZZKILL - STRENGTH RUNE GRABS
-        "xv %i yv %i string2 \"     HA:%i\" "   // BUZZKILL - HASTE RUNE GRABS
-        "xv %i yv %i string2 \"RG:%i\" "        // BUZZKILL - REGEN RUNE GRABS
-        "xv %i yv %i string2 \"     RS:%i\" "   // BUZZKILL - RESIST RUNE GRABS
-
-        "xv %i yv %i string2 \"FC:%s\" "
-        "xv %i yv %i string2 \"ST:%i\" "
-        "xv %i yv %i string2 \"     HA:%i\" "
-        "xv %i yv %i string2 \"RG:%i\" "
-        "xv %i yv %i string2 \"     RS:%i\" "
-
-        "xv %i yv %i num 4 19 "
-        "xv %i yv %i num 4 20 ",
+        //"xv %i yv %i num 4 24 "               // BUZZKILL - BLUE CAPS     // NOT NEEDED - IN v6.1 HUD
+        //"xv %i yv %i num 4 25 "               // BUZZKILL - RED CAPS      // NOT NEEDED - IN v6.1 HUD
+            
+        //"xv %i yv %i num 4 19 "               // BUZZKILL - BLUE SCORE    // NOT NEEDED - IN v6.1 HUD
+        //"xv %i yv %i num 4 20 ",              // BUZZKILL - RED SCORE     // NOT NEEDED - IN v6.1 HUD
 
         0, 0, "redlion_i",
         160, 0, "bluewolf_i",
         32, 0, "redtag",
         192, 0, "bluetag",
 
-        0, -32, "redlion_i",
-        160, -32, "bluewolf_i",
-        32, -32, "redtag",
-        192, -32, "bluetag",
+        36, 0,
+        36, 8, redfc,
+        36, 16,
+        36, 24, red_runes,
 
-        36, -30, redcaps,
-        36, -20, red_item_quad,
-        52, -20, red_item_shield,
-        36, -10, red_item_armor,
-        52, -10, red_item_mega,
+        196, 0,
+        196, 8, bluefc,
+        196, 16,
+        196, 24, blue_runes
 
-        196, -30, bluecaps,
-        196, -20, blue_item_quad,
-        212, -20, blue_item_shield,
-        196, -10, blue_item_armor,
-        212, -10, blue_item_mega,
+        //90, 4,
+        //250, 4,
 
-        36, 2, redfc,
-        36, 12, red_rune_strength,
-        52, 12, red_rune_haste,
-        36, 22, red_rune_regen,
-        52, 22, red_rune_resist,
-
-        196, 2, bluefc,
-        196, 12, blue_rune_strength,
-        212, 12, blue_rune_haste,
-        196, 22, blue_rune_regen,
-        212, 22, blue_rune_resist,
-
-        90, -28,
-        250, -28
+        //90, -28,
+        //250, -28
 
         );
 
@@ -1201,6 +1301,549 @@ void CTFSquadboardMessage (edict_t *ent, edict_t* killer) // ADC
 	gi.WriteString (string);
 }
 
+/*
+==================
+Statboard
+
+Draw instead of help message.
+Note that it isn't that hard to overflow the 1400 byte message limit!
+==================
+*/
+void Statboard(edict_t* ent)
+{
+    StatboardMessage(ent, ent->enemy);
+    gi.unicast(ent, true);
+}
+
+/*
+==================
+StatboardMessage
+
+==================
+*/
+void StatboardMessage(edict_t* ent, edict_t* killer)
+{
+    char    string[MAX_MSGLEN];
+    char    string2[MAX_MSGLEN];
+    int     blue, red;
+
+    size_t  stringlength;
+    int     i;
+    size_t  j;
+    int     k;
+
+    int     redsorted[MAX_CLIENTS];
+    int     redsortedscores[MAX_CLIENTS];
+    int     bluesorted[MAX_CLIENTS];
+    int     bluesortedscores[MAX_CLIENTS];
+
+    int     score;
+
+    int     rune_strength;
+    int     rune_haste;
+    int     rune_regen;
+    int     rune_resist;
+    int     item_mega;
+    int     item_armor;
+    int     item_shield;
+    int     item_quad;
+
+    int     x, y;
+
+    gclient_t*  cl;
+    edict_t*    cl_ent;
+
+    blue = 0;
+    red = 0;
+    
+    // sort the clients by score
+    for (i = 0; i < game.maxclients; i++)
+    {
+        cl_ent = g_edicts + 1 + i;
+        cl = &game.clients[i];
+
+        if (!cl_ent->inuse)
+            continue;
+
+        score = stats_get(cl_ent, STATS_SCORE);
+
+        rune_strength = stats_get(cl_ent, STATS_RUNE_STRENGTH);
+        rune_haste = stats_get(cl_ent, STATS_RUNE_HASTE);
+        rune_regen = stats_get(cl_ent, STATS_RUNE_REGEN);
+        rune_resist = stats_get(cl_ent, STATS_RUNE_RESIST);
+        item_quad = stats_get(cl_ent, STATS_ITEM_QUAD);
+        item_shield = stats_get(cl_ent, STATS_ITEM_SHIELD);
+        item_armor = stats_get(cl_ent, STATS_ITEM_ARMOR);
+        item_mega = stats_get(cl_ent, STATS_ITEM_MEGA);
+
+        if (cl_ent->client->ctf.teamnum == CTF_TEAM_RED)
+        {
+
+            for (j = 0; j < red; j++)
+            {
+                if (score > redsortedscores[j])
+                    break;
+            }
+            for (k = red; k > j; k--)
+            {
+                redsorted[k] = redsorted[k - 1];
+                redsortedscores[k] = redsortedscores[k - 1];
+            }
+            redsorted[j] = i;
+            redsortedscores[j] = score;
+            red++;
+        }
+        else if (cl_ent->client->ctf.teamnum == CTF_TEAM_BLUE)
+        {
+
+            for (j = 0; j < blue; j++)
+            {
+                if (score > bluesortedscores[j])
+                    break;
+            }
+            for (k = blue; k > j; k--)
+            {
+                bluesorted[k] = bluesorted[k - 1];
+                bluesortedscores[k] = bluesortedscores[k - 1];
+            }
+            bluesorted[j] = i;
+            bluesortedscores[j] = score;
+            blue++;
+        }
+
+    }
+
+    string[0] = 0;
+    string2[0] = 0;
+    stringlength = 0;
+    y = 32 * 8;
+
+    // DRAW STATBOARD AND ADD TEAM STATS
+    {
+        Com_sprintf(string2, sizeof(string2),
+            "xv %i yv %i picn %s "
+            "xv %i yv %i picn %s ",
+
+            -102, -35, "pb",
+            -102, -27, "pt"
+
+        );
+        
+        j = strlen(string2);
+        if (stringlength + j < 1024)
+        {
+            strcpy(string + stringlength, string2);
+            stringlength += j;
+        }
+    }
+
+    for (i = 0; i < red; i++)
+    {
+        cl = &game.clients[redsorted[i]];
+        cl_ent = g_edicts + 1 + redsorted[i];
+
+        x = -91;
+        y = 34 + 8 * i;
+
+        // send the layout        
+        Com_sprintf(string2, sizeof(string2),
+            "xv %i yv %i string2 \"%-15s %3i %2i %2i %2i\" ",
+
+            x, y,
+            cl->pers.netname,
+            stats_get(cl_ent, STATS_FRAGS),
+            stats_get(cl_ent, STATS_OFFENSE_CARRIER),
+            stats_get(cl_ent, STATS_DEFENSE_FLAG) + stats_get(cl_ent, STATS_DEFENSE_BASE),
+            stats_get(cl_ent, STATS_RETURNS)
+
+        );
+
+        j = strlen(string2);
+        if (stringlength + j > 1024)
+            break;
+        strcpy(string + stringlength, string2);
+        stringlength += j;
+
+    }
+
+    for (i = 0; i < blue; i++)
+    {
+        cl = &game.clients[bluesorted[i]];
+        cl_ent = g_edicts + 1 + bluesorted[i];
+
+        x = 171;
+        y = 34 + 8 * i;
+
+        Com_sprintf(string2, sizeof(string2),
+            "xv %i yv %i string2 \"%-15s %3i %2i %2i %2i\" ",
+
+            x, y,
+            cl->pers.netname,
+            stats_get(cl_ent, STATS_SCORE),            
+            stats_get(cl_ent, STATS_CAPTURES),
+            stats_get(cl_ent, STATS_DEFENSE_FLAG) + stats_get(cl_ent, STATS_DEFENSE_BASE),
+            stats_get(cl_ent, STATS_RETURNS)
+        
+        );
+
+
+
+        j = strlen(string2);
+        if (stringlength + j > 1024)
+            break;
+        strcpy(string + stringlength, string2);
+        stringlength += j;
+
+    }
+
+    gi.WriteByte(svc_layout);
+    gi.WriteString(string);
+
+}
+
+/*
+==================
+TeamStatboard
+
+Draw instead of help message.
+Note that it isn't that hard to overflow the 1400 byte message limit!
+==================
+*/
+void TeamStatboard(edict_t* ent)
+{
+    TeamStatboardMessage(ent, ent->enemy);
+    gi.unicast(ent, true);
+}
+
+/*
+==================
+TeamStatboardMessage
+
+==================
+*/
+void TeamStatboardMessage(edict_t* ent, edict_t* killer)
+{
+    char    string[MAX_MSGLEN];
+    char    string2[MAX_MSGLEN];
+    int     blue, red;
+
+    int     blue_rune_strength = 0;
+    int     blue_rune_haste = 0;
+    int     blue_rune_regen = 0;
+    int     blue_rune_resist = 0;
+    int     blue_item_mega = 0;
+    int     blue_item_quad = 0;
+    int     blue_item_armor = 0;
+    int     blue_item_shield = 0;
+    int     red_rune_strength = 0;
+    int     red_rune_haste = 0;
+    int     red_rune_regen = 0;
+    int     red_rune_resist = 0;
+    int     red_item_mega = 0;
+    int     red_item_quad = 0;
+    int     red_item_armor = 0;
+    int     red_item_shield = 0;
+
+    size_t  stringlength;
+    int     i;
+    size_t  j;
+    int     k;
+
+    int     redsorted[MAX_CLIENTS];
+    int     redsortedscores[MAX_CLIENTS];
+    int     bluesorted[MAX_CLIENTS];
+    int     bluesortedscores[MAX_CLIENTS];
+
+    int     score;
+
+    int     rune_strength;
+    int     rune_haste;
+    int     rune_regen;
+    int     rune_resist;
+    int     item_mega;
+    int     item_armor;
+    int     item_shield;
+    int     item_quad;
+
+    //int     x;
+    int     y;
+
+    gclient_t* cl;
+    edict_t* cl_ent;
+
+    blue = 0;
+    red = 0;
+
+    // sort the clients by score
+    for (i = 0; i < game.maxclients; i++)
+    {
+        cl_ent = g_edicts + 1 + i;
+        cl = &game.clients[i];
+
+        if (!cl_ent->inuse)
+            continue;
+
+        score = stats_get(cl_ent, STATS_SCORE);
+
+        rune_strength = stats_get(cl_ent, STATS_RUNE_STRENGTH);
+        rune_haste = stats_get(cl_ent, STATS_RUNE_HASTE);
+        rune_regen = stats_get(cl_ent, STATS_RUNE_REGEN);
+        rune_resist = stats_get(cl_ent, STATS_RUNE_RESIST);
+        item_quad = stats_get(cl_ent, STATS_ITEM_QUAD);
+        item_shield = stats_get(cl_ent, STATS_ITEM_SHIELD);
+        item_armor = stats_get(cl_ent, STATS_ITEM_ARMOR);
+        item_mega = stats_get(cl_ent, STATS_ITEM_MEGA);
+
+        if (cl_ent->client->ctf.teamnum == CTF_TEAM_RED)
+        {
+            red_rune_strength += rune_strength;
+            red_rune_haste += rune_haste;
+            red_rune_regen += rune_regen;
+            red_rune_resist += rune_resist;
+            red_item_quad += item_quad;
+            red_item_shield += item_shield;
+            red_item_armor += item_armor;
+            red_item_mega += item_mega;
+
+            for (j = 0; j < red; j++)
+            {
+                if (score > redsortedscores[j])
+                    break;
+            }
+            for (k = red; k > j; k--)
+            {
+                redsorted[k] = redsorted[k - 1];
+                redsortedscores[k] = redsortedscores[k - 1];
+            }
+            redsorted[j] = i;
+            redsortedscores[j] = score;
+            red++;
+        }
+        else if (cl_ent->client->ctf.teamnum == CTF_TEAM_BLUE)
+        {
+            blue_rune_strength += rune_strength;
+            blue_rune_haste += rune_haste;
+            blue_rune_regen += rune_regen;
+            blue_rune_resist += rune_resist;
+            blue_item_quad += item_quad;
+            blue_item_shield += item_shield;
+            blue_item_armor += item_armor;
+            blue_item_mega += item_mega;
+
+            for (j = 0; j < blue; j++)
+            {
+                if (score > bluesortedscores[j])
+                    break;
+            }
+            for (k = blue; k > j; k--)
+            {
+                bluesorted[k] = bluesorted[k - 1];
+                bluesortedscores[k] = bluesortedscores[k - 1];
+            }
+            bluesorted[j] = i;
+            bluesortedscores[j] = score;
+            blue++;
+        }
+
+    }
+
+    string[0] = 0;
+    string2[0] = 0;
+    stringlength = 0;
+    y = 32 * 8;
+
+    // DRAW TEAMSTATBOARD AND ADD TEAM STATS
+    {
+        Com_sprintf(string2, sizeof(string2),
+            "xv %i yv %i picn %s "
+            "xv %i yv %i picn %s "
+
+            "xv %i yv %i string2 \"%i\" "   // BUZZKILL - RED TEAM QUAD GRABS
+            "xv %i yv %i string2 \"%i\" "   // BUZZKILL - RED TEAM SHIELD GRABS
+            "xv %i yv %i string2 \"%i\" "   // BUZZKILL - RED TEAM RED ARMOR GRABS
+            "xv %i yv %i string2 \"%i\" "   // BUZZKILL - RED TEAM MEGA HEALTH GRABS
+
+            "xv %i yv %i string2 \"%i\" "
+            "xv %i yv %i string2 \"%i\" "
+            "xv %i yv %i string2 \"%i\" "
+            "xv %i yv %i string2 \"%i\" "
+
+            "xv %i yv %i string2 \"%i\" "   // BUZZKILL - RED TEAM STRENGTH RUNE GRABS
+            "xv %i yv %i string2 \"%i\" "   // BUZZKILL - RED TEAM HASTE RUNE GRABS
+            "xv %i yv %i string2 \"%i\" "   // BUZZKILL - RED TEAM RESIST RUNE GRABS
+            "xv %i yv %i string2 \"%i\" "   // BUZZKILL - RED TEAM REGEN RUNE GRABS
+
+            "xv %i yv %i string2 \"%i\" "
+            "xv %i yv %i string2 \"%i\" "
+            "xv %i yv %i string2 \"%i\" "
+            "xv %i yv %i string2 \"%i\" ",
+
+            -102, -35, "tb",
+            -102, -27, "tt",
+
+            -54, -19, red_item_quad,
+            -54, -10, red_item_shield,
+            -54, -1, red_item_armor,
+            -54, 8, red_item_mega,
+
+            209, -19, blue_item_quad,
+            209, -10, blue_item_shield,
+            209, -1, blue_item_armor,
+            209, 8, blue_item_mega,
+
+            10, -19, red_rune_strength,
+            10, -10, red_rune_haste,
+            10, -1, red_rune_resist,
+            10, 8, red_rune_regen,
+
+            273, -19, blue_rune_strength,
+            273, -10, blue_rune_haste,
+            273, -1, blue_rune_resist,
+            273, 8, blue_rune_regen
+
+        );
+
+        j = strlen(string2);
+        if (stringlength + j < 1024)
+        {
+            strcpy(string + stringlength, string2);
+            stringlength += j;
+        }
+    }
+
+    gi.WriteByte(svc_layout);
+    gi.WriteString(string);
+
+}
+
+/*
+==================
+Railboard
+
+Draw instead of help message.
+Note that it isn't that hard to overflow the 1400 byte message limit!
+==================
+*/
+void Railboard(edict_t* ent)
+{
+    RailboardMessage(ent, ent->enemy);
+    gi.unicast(ent, true);
+}
+
+/*
+==================
+RailboardMessage
+
+==================
+*/
+void RailboardMessage(edict_t* ent, edict_t* killer)
+{
+    char    string[MAX_MSGLEN];
+    char    string2[MAX_MSGLEN];
+
+    size_t  stringlength;
+    int     i;
+    size_t  j;
+    int     k;
+    int     player = 0;
+    int     rails = 0;
+    int     playersorted[MAX_CLIENTS];
+    int     playersortedrails[MAX_CLIENTS];
+
+    int     x, y;
+
+    gclient_t* cl;
+    edict_t* cl_ent;
+
+    // sort the clients by rail kills
+    for (i = 0; i < game.maxclients; i++)
+    {
+        cl_ent = g_edicts + 1 + i;
+        cl = &game.clients[i];
+
+        if (!cl_ent->inuse)
+            continue;
+
+        rails = stats_get(cl_ent, STATS_RAIL_KILL);
+
+        if (cl_ent->client->ctf.teamnum == CTF_TEAM_BLUE || cl_ent->client->ctf.teamnum == CTF_TEAM_RED)
+        {
+            for (j = 0; j < player; j++)
+            {
+                if (rails > playersortedrails[j])
+                    break;
+            }
+            for (k = player; k > j; k--)
+            {
+                playersorted[k] = playersorted[k - 1];
+                playersortedrails[k] = playersortedrails[k - 1];
+            }
+            playersorted[j] = i;
+            playersortedrails[j] = rails;
+            player++;
+        }
+
+    }
+
+    string[0] = 0;
+    string2[0] = 0;
+    stringlength = 0;
+    y = 32 * 8;
+
+    // DRAW STATBOARD AND ADD RAILS
+    {
+        Com_sprintf(string2, sizeof(string2),
+            "xv %i yv %i picn %s "
+            "xv %i yv %i picn %s ",
+
+            29, -35, "rb",
+            41, -26, "rt"
+
+        );
+
+        j = strlen(string2);
+        if (stringlength + j < 1024)
+        {
+            strcpy(string + stringlength, string2);
+            stringlength += j;
+        }
+    }
+
+    for (i = 0; i < player; i++)
+    {
+        cl = &game.clients[playersorted[i]];
+        cl_ent = g_edicts + 1 + playersorted[i];
+
+        x = 40;
+        y = -18 + 8 * i;
+
+        // send the layout        
+        Com_sprintf(string2, sizeof(string2),
+            "xv %i yv %i string2 \"%-15s %2i %2i %3i %3i\" ",
+
+            x, y,
+            cl->pers.netname,
+            stats_get(cl_ent, STATS_RAIL_KILL),
+            stats_get(cl_ent, STATS_RAIL_HIT),
+            stats_get(cl_ent, STATS_RAIL_SHOT),
+            cl_ent->client->p_stats_player->stats[STATS_RAIL_SHOT] == 0 ? 0 : 100 * 
+            cl_ent->client->p_stats_player->stats[STATS_RAIL_HIT] / 
+            cl_ent->client->p_stats_player->stats[STATS_RAIL_SHOT]
+        );
+
+        j = strlen(string2);
+        if (stringlength + j > 1024)
+            break;
+        strcpy(string + stringlength, string2);
+        stringlength += j;
+
+    }
+
+    gi.WriteByte(svc_layout);
+    gi.WriteString(string);
+
+}
+
 /*
 ==================
 Cmd_Score_f
@@ -1216,6 +1859,9 @@ void Cmd_Score_f (edict_t *ent)
     ent->client->showmod = false;
     ent->client->showmenu = false;
 	ent->client->showsquadboard = false; // ADC
+    ent->client->showstatboard = false; // BUZZKILL
+    ent->client->showteamstatboard = false; // BUZZKILL
+    ent->client->showrailboard = false; // BUZZKILL
 
 	if (!deathmatch->value && !coop->value)
 		return;
@@ -1261,6 +1907,105 @@ void Cmd_Squadboard_f (edict_t *ent)
 }
 // ADC
 
+// BUZZKILL
+/*
+==================
+Cmd_Statboard_f
+
+Display the statboard
+==================
+*/
+void Cmd_Statboard_f(edict_t* ent)
+{
+    ent->client->showhelp = false;
+    ent->client->showinventory = false;
+    ent->client->showctfhud = false;
+    ent->client->showmod = false;
+    ent->client->showmenu = false;
+    ent->client->showscores = false;
+    ent->client->showsquadboard = false;
+    ent->client->showteamstatboard = false;
+    ent->client->showrailboard = false;
+
+    if (!deathmatch->value && !coop->value)
+        return;
+
+    if (ent->client->showstatboard)
+    {
+        ent->client->showstatboard = false;
+        return;
+    }
+
+    ent->client->showstatboard = true;
+    Statboard(ent);
+}
+
+/*
+==================
+Cmd_TeamStatboard_f
+
+Display the teamstatboard
+==================
+*/
+void Cmd_TeamStatboard_f(edict_t* ent)
+{
+    ent->client->showinventory = false;
+    ent->client->showhelp = false;
+    ent->client->showctfhud = false;
+    ent->client->showmod = false;
+    ent->client->showmenu = false;
+    ent->client->showsquadboard = false; // ADC
+    ent->client->showstatboard = false; // BUZZKILL
+    ent->client->showscores = false; // BUZZKILL
+    ent->client->showrailboard = false; // BUZZKILL
+
+    if (!deathmatch->value && !coop->value)
+        return;
+
+    if (ent->client->showteamstatboard)
+    {
+        ent->client->showteamstatboard = false;
+        return;
+    }
+
+    ent->client->showteamstatboard = true;
+    TeamStatboard(ent);
+}
+
+/*
+==================
+Cmd_Railboard_f
+
+Display the railboard
+==================
+*/
+void Cmd_Railboard_f(edict_t * ent)
+{
+    ent->client->showhelp = false;
+    ent->client->showinventory = false;
+    ent->client->showctfhud = false;
+    ent->client->showmod = false;
+    ent->client->showmenu = false;
+    ent->client->showscores = false;
+    ent->client->showsquadboard = false;
+    ent->client->showstatboard = false;
+    ent->client->showteamstatboard = false;
+
+    if (!deathmatch->value && !coop->value)
+        return;
+
+    if (ent->client->showrailboard)
+    {
+        ent->client->showrailboard = false;
+        return;
+    }
+
+    ent->client->showrailboard = true;
+    Railboard(ent);
+}
+
+// BUZZKILL
+
 /*
 ==================
 HelpComputer
@@ -1327,6 +2072,9 @@ void Cmd_Help_f (edict_t *ent)
     ent->client->showmod = false;
     ent->client->showmenu = false;
 	ent->client->showsquadboard = false; // ADC
+    ent->client->showstatboard = false; // BUZZKILL
+    ent->client->showteamstatboard = false; // BUZZKILL
+    ent->client->showrailboard = false; // BUZZKILL
 
 	if (ent->client->showhelp && (ent->client->pers.game_helpchanged == game.helpchanged))
 	{
@@ -1640,7 +2388,8 @@ edict_t     *cl_ent;
         if (ent->client->pers.health <= 0 || level.intermissiontime
             || ent->client->showscores || ent->client->showctfhud 
             || ent->client->showmod || ent->client->showmenu
-			|| ent->client->showsquadboard) // ADC
+			|| ent->client->showsquadboard || ent->client->showstatboard
+            || ent->client->showteamstatboard || ent->client->showrailboard) // ADC //BUZZKILL
             ent->client->ps.stats[STAT_LAYOUTS] |= 1;
         if (ent->client->showinventory && ent->client->pers.health > 0)
             ent->client->ps.stats[STAT_LAYOUTS] |= 2;
@@ -1649,7 +2398,8 @@ edict_t     *cl_ent;
     {
         if (ent->client->showscores || ent->client->showhelp
             || ent->client->showctfhud || ent->client->showmod
-			|| ent->client->showsquadboard) // ADC
+			|| ent->client->showsquadboard || ent->client->showstatboard
+            || ent->client->showteamstatboard || ent->client->showrailboard) // ADC // BUZZKILL
             ent->client->ps.stats[STAT_LAYOUTS] |= 1;
         if (ent->client->showinventory && ent->client->pers.health > 0)
             ent->client->ps.stats[STAT_LAYOUTS] |= 2;
@@ -1705,7 +2455,7 @@ void G_SetSpectatorStats (edict_t *ent)
 	
 	//bat - I think that cl->pers.health is only supposed to be checked in deathmatch
 	
-	if(level.intermissiontime || cl->showscores || cl->showmenu)
+	if(level.intermissiontime || cl->showscores || cl->showmenu || cl->showrailboard || cl->showstatboard || cl->showteamstatboard)
 		cl->ps.stats[STAT_LAYOUTS] |= 1;
 	if (cl->showinventory && cl->pers.health > 0)
 		cl->ps.stats[STAT_LAYOUTS] |= 2;
diff --git c/p_stats.c w/p_stats.c
index 33a49cb..2c5d8fb 100644
--- c/p_stats.c
+++ w/p_stats.c
@@ -225,6 +225,20 @@ void stats_output(edict_t* ent, stats_player_s* p_player)
 		p_player->stats[STATS_ITEM_ARMOR],
 		p_player->stats[STATS_ITEM_MEGA]);
 	strcat(outbuf, tmpbuf);
+
+	sprintf(tmpbuf, "--RAIL---------------------------------------\nShots=%ld Hits=%ld Kills=%ld Accuracy=%ld\n---------------------------------------------\n",
+		p_player->stats[STATS_RAIL_SHOT],
+		p_player->stats[STATS_RAIL_HIT],
+		p_player->stats[STATS_RAIL_KILL],
+		p_player->stats[STATS_RAIL_SHOT] == 0 ? 0 : 100 * p_player->stats[STATS_RAIL_HIT] / p_player->stats[STATS_RAIL_SHOT]);
+	strcat(outbuf, tmpbuf);
+
+	sprintf(tmpbuf, "--DAMAGE-------------------------------------\nGiven=%ld Received=%ld Eff=%ld\n---------------------------------------------\n",
+		p_player->stats[STATS_DAMAGE_GIVEN],
+		p_player->stats[STATS_DAMAGE_REC],
+		p_player->stats[STATS_DAMAGE_REC] == 0 ? 100 : 100 * p_player->stats[STATS_DAMAGE_GIVEN] / p_player->stats[STATS_DAMAGE_REC]);
+	strcat(outbuf, tmpbuf);
+
 	// BUZZKILL - IMPROVED ANALYTICS - END
 
 	ctf_SafePrint(ent, PRINT_HIGH, outbuf);
diff --git c/p_stats.h w/p_stats.h
index 19e7bd4..fb7087c 100644
--- c/p_stats.h
+++ w/p_stats.h
@@ -30,9 +30,19 @@
 #define STATS_ITEM_ARMOR		22	// picked up red armor
 #define STATS_ITEM_MEGA			23  // picked up mega health
 #define STATS_IS_FC				24	// is the flag carrier
+#define STATS_HAS_ST			25	// has the strength rune
+#define STATS_HAS_RS			26	// has the resist rune
+#define STATS_HAS_HA			27	// has the haste rune
+#define STATS_HAS_RG			29	// has the regen rune
+#define STATS_RAIL_SHOT			30  // railgun shots fired
+#define STATS_RAIL_HIT			31  // railgun hits
+#define STATS_RAIL_KILL			32  // railgun kills (not equal to hits due to armor, runes, etc.)
+#define STATS_RAIL_ACCURACY		33  // railgun accuracy
+#define STATS_DAMAGE_GIVEN		34  // damage given
+#define STATS_DAMAGE_REC		35  // damage received
 // BUZZKILL - IMPROVED ANALYTICS - END
 
-#define MAX_PLAYER_STATS		25	
+#define MAX_PLAYER_STATS		36	
 
 typedef struct {
 	char name[MAX_INFO_STRING];
diff --git c/p_view.c w/p_view.c
index b3b488b..c13cd32 100644
--- c/p_view.c
+++ w/p_view.c
@@ -1118,6 +1118,34 @@ void ClientEndServerFrame (edict_t *ent)
 		}
 	}
 // ADC
+
+// BUZZKILL
+	else if (ent->client->showstatboard && deathmatch->value)
+	{
+		if (!(level.framenum & 31))
+		{
+			StatboardMessage(ent, ent->enemy);
+			gi.unicast(ent, false);
+		}
+	}
+	else if (ent->client->showteamstatboard && deathmatch->value)
+	{
+		if (!(level.framenum & 31))
+		{
+			TeamStatboardMessage(ent, ent->enemy);
+			gi.unicast(ent, false);
+		}
+	}
+	else if (ent->client->showrailboard && deathmatch->value)
+	{
+		if (!(level.framenum & 31))
+		{
+			RailboardMessage(ent, ent->enemy);
+			gi.unicast(ent, false);
+		}
+	}
+
+// BUZZKILL
 	else if (ent->client->showmod)
 	{
 		if (!(level.framenum & 7))
@@ -1257,8 +1285,9 @@ int ClientShowID(edict_t *ent, char * buf)
 
 	if (ent->client->showscores || ent->client->showhelp ||
 		ent->client->showinventory || ent->client->showmod ||
-		ent->client->showmenu
-		|| ent->client->showsquadboard) // ADC
+		ent->client->showmenu || ent->client->showsquadboard ||
+		ent->client->showstatboard || ent->client->showteamstatboard ||
+		ent->client->showrailboard) // ADC // BUZZKILL
 		return 0;
 
 	// only update every second for each player
